core/common/src/main/kotlin/com/google/samples/apps/nowinandroid/core/navigation/NiaNavigator.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.nowinandroid.core.navigation
+
+/**
+ * Provides [NiaBaseNavigator] instances for routes/destinations.
+ * Individual [NiaBaseNavigator] instances can then be used to create or navigate to destinations.
+ */
+interface NiaNavigator {
```

Comment:
This should ideally be called `NavigatorProvider`. This + multi-binding lets us not have to inject instances for each route's navigator.